### ISSUE TITLE
Feature Request: Insert values for front matter of notes

### ISSUE THREAD

**Issue by kazdonkai** [RafaelGB/obsidian-db-folder#39]
I have filled in the Front Matter values for “book title”, “subtitle”, “author”, “publisher”, etc.
Since it is useless to re-fill the database, could you please consider adding an option to read the values from Front Matter and automatically fill in the columns if the column name is set to "book title”, etc.

**Comment by RafaelGB** [comment]
Hello @kazdonkai  Thanks for your suggest.
Right now you have to change the Label of existed columns or add new ones and change it too.

I am afraid that names with spaces are not supported ("book title") inside your front matter. As a label it is perfectly possible, but the label is trimmed to be used "booktitle" as a key

I will consider your petition to create a template selection.

The idea would be to add within the database configuration, the selection of a file and create the columns based on its fields, both frontmatter and inline.

About the keys with spaces, I will check if support it involves a lot of refactoring effort

**Comment by imeed166** [comment]
+1 would also love for the database to autopopulate based on the fields present.

**Comment by RafaelGB** [comment]
need some details but already works

https://user-images.githubusercontent.com/11924043/173366911-aff57162-d8b6-4167-9aa3-88bf20bedde2.mov

**Comment by imeed166** [comment]
Nice work, can we choose from all files in the vault ?

**Comment by RafaelGB** [comment]
yes, but I want to filter it by the folder files, dunno whats the best option.

**Comment by imeed166** [comment]
If it were me, I wouldn't let the user choose a file, I would scan all the files that are included in the source (folder, tag, link) and include all the metadata directly.

Wouldn't that be simpler ?

**Comment by RafaelGB** [comment]
that will be another option, just take all fields. But if you just want a reference, I think the file selector should be filtered by the folder content

**Comment by imeed166** [comment]
Ah, okay, then yeah, you're right, only show the files that are included in the source.

**Comment by RafaelGB** [comment]
https://github.com/RafaelGB/obsidian-db-folder/releases/tag/1.7.1 check new version =) this issue will be available

**Comment by kazdonkai** [comment]
Adding from a template works very well.
Thanks.
